packages/react-instantsearch-core/src/connectors/connectConfigure.js

```diff
@@ -22,8 +22,9 @@ export default createConnector({
   transitionState(props, prevSearchState, nextSearchState) {
     const id = getId();
     const { children, contextValue, indexContextValue, ...items } = props;
+    const propKeys = Object.keys(props);
     const nonPresentKeys = this._props
```

Comment:
The name of the variable is confusing, it looks like we already have the non present keys. `previousPropsKeys` sounds a better name. We can also perform the call to `Object.keys` at this stage to be sync with `propsKeys` (which should be renamed to `propsKeys` notice the `S`).